Compilation failing on archlinux

I am seeing an error while compiling on archlinux. It looks like it might be fixed if `-fexceptions` is passed to the compiler.

I'm not too familiar with cmake so I'm not quite sure how to fix this, but I'm hoping the solution is pretty straightforward.

A complete log from running the setup script is attached.

And thank you so much for maintaining this. I have been using this setup script for over a year to play netplay and I really appreciate the work you've put in to keep things working on linux :smile:

[log.txt](https://github.com/FasterMelee/FasterMelee-installer/files/2431889/log.txt)

Same build failure on Arch.

...
[ 57%] Building CXX object Externals/wxWidgets3/CMakeFiles/wx.dir/src/unix/stdpaths.cpp.o
[ 57%] Building CXX object Externals/wxWidgets3/CMakeFiles/wx.dir/src/unix/uiactionx11.cpp.o
[ 57%] Building CXX object Externals/wxWidgets3/CMakeFiles/wx.dir/src/unix/utilsx11.cpp.o
[ 57%] Linking CXX static library libwx.a
[ 57%] Built target wx
make: *** [Makefile:163: all] Error 2

Hey guys! I've been really busy with school so I probably won't have time
to patch soundtouch, but I'll add the cmake flag to the installation script
later. Thanks for all the reports!

EDIT: The offending line appears in the master branch of Dolphin as well. For now, I suggest doing what jangler did up above.

On Mon, Oct 1, 2018 at 7:42 PM z0z0z <notifications@github.com> wrote:

> Same build failure on Arch.
>
> ...
> [ 57%] Building CXX object
> Externals/wxWidgets3/CMakeFiles/wx.dir/src/unix/stdpaths.cpp.o
> [ 57%] Building CXX object
> Externals/wxWidgets3/CMakeFiles/wx.dir/src/unix/uiactionx11.cpp.o
> [ 57%] Building CXX object
> Externals/wxWidgets3/CMakeFiles/wx.dir/src/unix/utilsx11.cpp.o
> [ 57%] Linking CXX static library libwx.a
> [ 57%] Built target wx
> make: *** [Makefile:163: all] Error 2
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/FasterMelee/FasterMelee-installer/issues/50#issuecomment-426099478>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEGT7hlfOTO_vOUuiQx2jMr2CAfcii3sks5ugqhrgaJpZM4XBGnb>
> .
>

--- a/setup
+++ b/setup
@@ -192,6 +192,11 @@ if [ "$FMVERSION" = "5.8.7" ]; then
 	sed -i "s|\${GIT_EXECUTABLE} describe --always --long --dirty|echo FM v$FMVERSION BETA|g" CMakeLists.txt # ensures compatibility w/ netplay
 	sed -i "s|\${GIT_EXECUTABLE} rev-parse --abbrev-ref HEAD|echo HEAD|g" CMakeLists.txt
 	# ---
+	
+	# --- Patch for SoundTouch
+	echo "Using static soundtouch from Externals..."
+	sed -i 's|message("Using shared soundtouch")|add_subdirectory(Externals/soundtouch)\n\tinclude_directories(Externals)|g' CMakeLists.txt #uses external soundtouch lib
+	# ---
 else	
 	# --- Patch tarball to display correct hash to other netplay clients
 	echo "Patching tarball..."

